### ISSUE TITLE
unable to reproduce the results reported in paper

### ISSUE THREAD

**Issue by lyxlynn** [YWolfeee/Training-Free-Guidance#2]
It's a nice work, and I tried running the code from your GitHub but encountered some issues.

I followed the instructions in the appendix of your paper and tested it on Gaussian deblur and super-resolution tasks.

The paper mentions that the tests generate 256 samples, so I only modified the parameters in scripts/.sh to per_sample_batch_size=16, num_samples=256, and eval_batch_size=16 for running. 

The weights were downloaded locally from https://huggingface.co/google/ddpm-cat-256/tree/main, and I modified the path to point to the local directory.

However, the metrics I obtained from the test are:

Gaussian blur:  FID: 71.27 LPIPS:0.1755

Super-Resolution: FID: 96.87 LPIPS:0.3088

The results seem to differ significantly from the metrics reported in the paper：

Gaussian blur in paper:  FID: 64.5 LPIPS:0.150

Super-Resolution in paper: FID: 65.9 LPIPS:0.190

Is there something wrong with my implementation, or is it possible that the script parameters in the GitHub repository are not optimized, such as the values for rho, mu, etc.?

**Comment by lyxlynn** [comment]
![image](https://github.com/user-attachments/assets/706950bd-0121-4ef0-87d9-6815bca0f554)
Hi,  I also tried the style transfer task. I set numer_samples=16 and target =./data/wikiart/2.png in scipts.style_transfer.sh. 
The visualization (the attached firgure) looks worse than yours in your paper and I got a strange results which are far way from the results in your paper:{'inception_score': 14.890585, 'style_score': -45.31602478027344, 'clip_score': 30.762971878051758}

**Comment by linhaowei1** [comment]
Hi,

Thank you for bringing this issue to our attention. After reviewing the scripts, I noticed that the `iter_steps` hyperparameter is missing. As mentioned in our paper, we set `iter_steps` to 4 in our experiments. However, the default value for `iter_steps` is 1. I tested both settings, and the evaluation results for Super Resolution are as follows:

`iter_steps = 1`:

```python
{'inception_score': 4.123416, 'fid': 92.75931733828293, 'neg_lpips': -0.3124462366104126}
```

`iter_steps = 4`:

```python
{'inception_score': 3.096159, 'fid': 66.17770272086358, 'neg_lpips': -0.1873655617237091}
```

The results seem to align with yours. Therefore, updating the scripts to explicitly set `iter_steps = 4` should make the results consistent with our paper. I’ll address this in an upcoming commit.

**Comment by lyxlynn** [comment]
Thanks a lot , I can get the correct results on SR tasks now with iter_steps=4.
However, there are still some problems about style_score metric in style transfer task. 
I got negative style score  when  evaluating 64 samples :
{'inception_score': 11.079268, 'style_score': -36.311553955078125, 'clip_score': 30.272268295288086}

Looking forward to your reply! Thanks again!
